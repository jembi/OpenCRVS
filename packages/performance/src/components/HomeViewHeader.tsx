import * as React from 'react'
import { IViewHeadingProps, ViewHeading } from './ViewHeading'
import { Header } from '@opencrvs/components/lib/interface'
import { ActionList } from '@opencrvs/components/lib/buttons'
import styled from '../styled-components'
import { TopMenu } from './TopMenu'
import { Logo, Offline, Online } from '@opencrvs/components/lib/icons'

const StretchedHeader = styled(Header)`
  justify-content: flex-end;
  padding-bottom: 50px;
  min-height: 280px;

  /* stylelint-disable */
  + ${ActionList} {
    /* stylelint-enable */
    margin-top: -80px;
  }
`

const StyledLogo = styled(Logo)`
  margin: -71px auto auto 17px;
`

const StyledOnline = styled.div`
  position: absolute;
  top: 32px;
  right: 234px;
`

const StyledOffline = styled.div`
  position: absolute;
  top: 32px;
  right: 234px;
`

export class HomeViewHeader extends React.Component<IViewHeadingProps> {
  render() {
    const { title, description, id } = this.props
    return (
      <StretchedHeader {...this.props}>
        {navigator.onLine ? (
          <StyledOnline>
            <Online />
          </StyledOnline>
        ) : (
          <StyledOffline>
            <Offline />
          </StyledOffline>
        )}
        <TopMenu hideBackButton={true} />
        <StyledLogo />
        <ViewHeading {...{ title, description, id }} />
        {this.props.children}
      </StretchedHeader>
    )
  }
}
