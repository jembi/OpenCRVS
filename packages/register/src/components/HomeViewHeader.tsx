import * as React from 'react'
import {
  IViewHeadingProps,
  ViewHeading
} from '@register/components/ViewHeading'
import { Header } from '@opencrvs/components/lib/interface'
import { ActionList } from '@opencrvs/components/lib/buttons'
import styled from '@register/styledComponents'
import { TopMenu } from '@register/components/TopMenu'
import { Logo } from '@opencrvs/components/lib/icons'
import ConnectivityStatus from '@register/components/ConnectivityStatus'
import { BodyContent } from '@opencrvs/components/lib/layout'

const HeaderWrapper = styled(Header)`
  display: block;
  justify-content: flex-end;
  padding-bottom: 50px;
  min-height: 280px;

  /* stylelint-disable */
  + ${ActionList} {
    /* stylelint-enable */
    margin-top: -80px;
  }
`

const StyledLogo = styled(Logo)`
  margin: -71px auto auto 20px;
`

export class HomeViewHeader extends React.Component<IViewHeadingProps> {
  render() {
    const { title, description, id } = this.props
    return (
      <HeaderWrapper>
        <BodyContent>
          <ConnectivityStatus />
          <TopMenu hideBackButton={true} />
          <StyledLogo />
          <ViewHeading {...{ title, description, id }} />
          {this.props.children}
        </BodyContent>
      </HeaderWrapper>
    )
  }
}
