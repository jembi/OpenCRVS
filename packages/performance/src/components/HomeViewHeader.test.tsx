import * as React from 'react'
import { createShallowRenderedComponent } from '../tests/util'
import { HomeViewHeader } from './HomeViewHeader'

describe('home view header component', () => {
  const testComponent = createShallowRenderedComponent(
    <HomeViewHeader
      id="test_heaader"
      title="Test title"
      description="Test description"
    />
  )

  it('renders without crashing', () => {
    expect(testComponent).toMatchSnapshot()
  })
})
