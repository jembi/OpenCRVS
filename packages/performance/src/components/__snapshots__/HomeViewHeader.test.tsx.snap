// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home view header component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeViewHeader
    description="Test description"
    id="test_heaader"
    title="Test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <Unknown />
        </styled.div>,
        <Connect(InjectIntl(TopMenuComponent))
          hideBackButton={true}
        />,
        <Styled(Component) />,
        <ViewHeading
          description="Test description"
          id="test_heaader"
          title="Test title"
        />,
        undefined,
      ],
      "description": "Test description",
      "id": "test_heaader",
      "title": "Test title",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "hideBackButton": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "description": "Test description",
          "id": "test_heaader",
          "title": "Test title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <Unknown />
          </styled.div>,
          <Connect(InjectIntl(TopMenuComponent))
            hideBackButton={true}
          />,
          <Styled(Component) />,
          <ViewHeading
            description="Test description"
            id="test_heaader"
            title="Test title"
          />,
          undefined,
        ],
        "description": "Test description",
        "id": "test_heaader",
        "title": "Test title",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "hideBackButton": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "Test description",
            "id": "test_heaader",
            "title": "Test title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
